### PR TITLE
ignore VersionNotFound in addition to ObjectNotFound while replicating

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
In replication, all objects are versioned, ignore version not found instead of object not found

## Motivation and Context
Fix wrong log reporting

## How to test this PR?
Not easy

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/bucket-replication.go [hunk 1]
```diff
@@ -1068,7 +1068,7 @@ func (ri ReplicateObjectInfo) replicateObject(ctx context.Context, objectAPI Obj
 		VersionSuspended: versionSuspended,
 	})
 	if err != nil {
-		if !isErrObjectNotFound(err) {
+		if !isErrVersionNotFound(err) {
```
Comment (klauspost) [inline cmd/bucket-replication.go]:
```suggestion
		if !isErrVersionNotFound(err) &&  !isErrObjectNotFound(err) {
```
Couldn't the object have been deleted? I don't see much bad in checking both.

File: cmd/bucket-replication.go [hunk 2]
```diff
@@ -1216,7 +1216,7 @@ func (ri ReplicateObjectInfo) replicateAll(ctx context.Context, objectAPI Object
...
```
Comment (klauspost) [inline cmd/bucket-replication.go]:
```suggestion
		if !isErrVersionNotFound(err) &&  !isErrObjectNotFound(err) {
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
formatting missing on the patch

```
cmd/bucket-replication.go:1071: File is not `gofmt`-ed with `-s` (gofmt)
		if !isErrVersionNotFound(err) &&  !isErrObjectNotFound(err) {
make: *** [Makefile:37: lint] Error 1
Error: Process completed with exit code 2.
```

@vadmeste

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**